rust_snuba/src/strategies/commit_log.rs

```diff
@@ -101,4 +211,83 @@ mod tests {
         assert_eq!(transformed.key, payload_clone.key);
         assert_eq!(transformed.payload, payload_clone.payload);
     }
+
+    #[test]
+    fn produce_commit_log() {
+        struct Noop {
+            pub payloads: Vec<KafkaPayload>,
+        }
+        impl ProcessingStrategy<KafkaPayload> for Noop {
+            fn poll(&mut self) -> Option<CommitRequest> {
+                None
+            }
+            fn submit(
+                &mut self,
+                message: Message<KafkaPayload>,
+            ) -> Result<(), MessageRejected<KafkaPayload>> {
+                self.payloads.push(message.payload());
+                Ok(())
+            }
+            fn close(&mut self) {}
+            fn terminate(&mut self) {}
+            fn join(&mut self, _timeout: Option<Duration>) -> Option<CommitRequest> {
+                None
+            }
+        }
+
+        struct MockProducer {
+            pub payloads: Vec<KafkaPayload>,
+        }
```

Comment:
I think we built an in memory producer for tests. You could use that one.

Reply:
Needs a refactor, will tackle it separately.

Reply:
Actually, it doesn't even exist. We need to build it.